tm2/pkg/bft/privval/state/state.go

```diff
@@ -0,0 +1,325 @@
+package state
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	tmtime "github.com/gnolang/gno/tm2/pkg/bft/types/time"
+	"github.com/gnolang/gno/tm2/pkg/errors"
+	osm "github.com/gnolang/gno/tm2/pkg/os"
+)
+
+// Step is the step in the consensus process.
+type Step uint8
+
+const (
+	StepNone      Step = 0 // Undefined step
```

Comment:
When is this even possible?

Reply:
When you initialize a new empty instance of the state struct, for example. t seems very common to me in Go to have an `Unknown` / `None` value at 0 in the context of an iota enum.

On the current master, when you initialize a node, the `priv_validator_state.json` contains this: 
```json
{
	"height": "0",
	"round": "0",
	"step": 0
}
```

Which looks fine to me to have a `None` step as our previous step.

Reply:
This is an unused constant, we can drop it

Reply:
Done https://github.com/gnolang/gno/pull/3554/commits/7baef0c919d7c6694dfa312dfecd4d247e1b26cb